exchanges/stream/websocket.go

```diff
@@ -410,11 +414,17 @@ func (w *Websocket) Connect() error {
 			break
 		}
 
+		w.connections[conn] = &w.connectionManager[i]
+		w.connectionManager[i].Connection = conn
+
+		if !conn.IsConnected() {
+			multiConnectFatalError = fmt.Errorf("%s websocket: [conn:%d] [URL:%s] failed to connect", w.exchangeName, i+1, conn.URL)
+			break
```

Comment:
Why in this order? Why not only set things when you know they're successful? Seems like this check should be before setting it